cupy/core/_kernel.pyx

```diff
@@ -187,7 +187,7 @@ cdef class _Args:
         for obj in objs:
             arg = Arg.from_obj(obj)
             # Check device of an ndarray
-            if arg.is_ndarray():
+            if isinstance(obj, ndarray):
```

Comment:
Please do not revert this line.
The purpose of `is_ndarray` is to avoid repeated `isinstance`.

Reply:
When `obj` is a cuda_array_interface object, `arg.is_ndarray()` will be `Ture` but this condition should be evaluated to `False`.

Reply:
Do you mean that because the line below would not work?
Then, let's change the line instead.

```diff
-                _check_array_device_id(obj, device_id)
+                _check_array_device_id(arg.obj, device_id)
```

Basically, `_cuda_array_interface__` handling should be confined in `Arg.from_obj` and no other code should  care about it.

Reply:
Fixed.